.circleci/config.yml

```diff
@@ -106,94 +156,358 @@ references:
+  
+  build_and_push_docker_image: &build_and_push_docker_image
+    run:
+      name: Build docker image
+      command: |
+        . build.sh circleci
+        mkdir -p workspace
+        echo $BUILD_TAG > workspace/build_tag
+
+  deploy_to_dev: &deploy_to_dev
+    run:
+      name: Deploy to Development environment
+      command: |
+        setup-kube-auth
+        kubectl config use-context development        
+        build_tag=`cat /tmp/workspace/build_tag`
+        ./deploy.sh $build_tag development circleci
 
-  # deploy_to_dev: &deploy_to_dev
-  #   run:
-  #     name: Deploy to dev environment
-  #     command: |
-  #       setup-kube-auth
-  #       kubectl config use-context development
-  #       sed -i -e s/:latest/:$CIRCLE_SHA1/ k8s-deploy/development/deployment.yaml
-  #       kubectl apply --record=false -f ./k8s-deploy/development/
-
-  # deploy_to_staging: &deploy_to_staging
-  #   run:
-  #     name: Deploy to staging environment
-  #     command: |
-  #       setup-kube-auth
-  #       kubectl config use-context staging
-  #       sed -i -e s/:latest/:$CIRCLE_SHA1/ k8s-deploy/staging/deployment.yaml
-  #       kubectl apply --record=false -f ./k8s-deploy/staging/
-  #
-  # deploy_to_production: &deploy_to_production
-  #   run:
-  #     name: Deploy to production environment
-  #     command: |
-  #       setup-kube-auth
-  #       kubectl config use-context production
-  #       sed -i -e s/:latest/:$CIRCLE_SHA1/ k8s-deploy/production/deployment.yaml
-  #       kubectl apply --record=false -f ./k8s-deploy/production/
+  deploy_to_staging: &deploy_to_staging
+    run:
+      name: Deploy to Staging environment
+      command: |
+        setup-kube-auth
+        kubectl config use-context staging       
+        build_tag=`cat /tmp/workspace/build_tag`
+        ./deploy.sh $build_tag staging circleci
+
+  deploy_to_qa: &deploy_to_qa
+    run:
+      name: Deploy to QA environment
+      command: |
+        setup-kube-auth
+        kubectl config use-context qa
+        build_tag=`cat /tmp/workspace/build_tag`
+        ./deploy.sh $build_tag qa circleci
+
+  deploy_to_demo: &deploy_to_demo
+    run:
+      name: Deploy to Demo environment
+      command: |
+        setup-kube-auth
+        kubectl config use-context demo
+        build_tag=`cat /tmp/workspace/build_tag`
+        ./deploy.sh $build_tag demo circleci
+
+  deploy_to_production: &deploy_to_production
+     run:
+       name: Deploy to production environment
+       command: |
+        setup-kube-auth
+        kubectl config use-context production
+        build_tag=`cat /tmp/workspace/build_tag`
+        ./deploy.sh $build_tag production circleci
 
 #  These are the jobs which run the steps. The jobs are used by the workflows.
 
 jobs:
-  build:
+  
+  # build_and_test_fake:
+  #   <<: *test_container_config
+  #   <<: *defaults   
+  #   steps:
+  #         - checkout
+  #         - setup_remote_docker:
+  #             docker_layer_caching: true
```

Comment:
Maybe we can get rid of this commented out bit?

Reply:
Done